src/infra/marketplace/marketplaceCmdRepo.go

```diff
@@ -437,6 +428,161 @@ func (repo *MarketplaceCmdRepo) InstallItem(
 	)
 }
 
+func (repo *MarketplaceCmdRepo) moveSelectedFiles(
+	sourceDir valueObject.UnixFilePath,
+	targetDir valueObject.UnixFilePath,
+	fileNames []valueObject.UnixFileName,
+	shouldKeepInstead bool,
+) error {
+	params := ""
+	if len(fileNames) > 0 {
+		fileNameFilterParams := "-name \"" + fileNames[0].String() + "\""
+		for _, fileToIgnore := range fileNames[1:] {
+			fileNameFilterParams += " -o -name \"" + fileToIgnore.String() + "\""
+		}
+		params = "\\( " + fileNameFilterParams + " \\)"
+	}
+
+	flags := []string{"-maxdepth 1"}
+	if shouldKeepInstead {
+		flags = append(flags, "-mindepth 1", "-not")
+	}
+	flagsStr := strings.Join(flags, " ")
+
+	moveCmd := fmt.Sprintf(
+		"find %s/ %s %s -exec mv -t %s {} +",
+		sourceDir.String(),
+		flagsStr,
+		params,
+		targetDir.String(),
+	)
+	_, err := infraHelper.RunCmdWithSubShell(moveCmd)
+	return err
+}
+
+func (repo *MarketplaceCmdRepo) uninstallSymlinkFilesRemoval(
+	installedItem entity.MarketplaceInstalledItem,
+	catalogItem entity.MarketplaceCatalogItem,
+	removalDestinationPath valueObject.UnixFilePath,
+) error {
+	installedHostnameStr := installedItem.Hostname.String()
+	installDirBackupPath, err := valueObject.NewUnixFilePath(
+		"/app/" + installedHostnameStr + "-backup",
+	)
+	if err != nil {
+		return err
+	}
+
+	installDirBackupPathStr := installDirBackupPath.String()
+	err = infraHelper.MakeDir(installDirBackupPathStr)
+	if err != nil {
+		return errors.New("CreateInstallDirBackupError: " + err.Error())
+	}
+
+	shouldKeepInstead := true
+	err = repo.moveSelectedFiles(
+		installedItem.InstallDirectory, installDirBackupPath,
+		catalogItem.UninstallFileNames, shouldKeepInstead,
+	)
+	if err != nil {
+		return errors.New("MoveNonUninstallFilesToBackupDirError: " + err.Error())
+	}
+
+	installedSlugStr := installedItem.Slug.String()
+	installUuidStr := installedItem.InstallUuid.String()
+	srcInstallDirPath := fmt.Sprintf(
+		"/app/%s-%s-%s",
+		installedSlugStr,
+		installedHostnameStr,
+		installUuidStr,
+	)
+	_, err = infraHelper.RunCmdWithSubShell(
+		"mv " + srcInstallDirPath + "/* " + removalDestinationPath.String(),
+	)
+	if err != nil {
+		return errors.New("MoveUninstallFilesToTrashError: " + err.Error())
+	}
+
+	_, err = infraHelper.RunCmdWithSubShell(
+		"rm -rf " + srcInstallDirPath,
+	)
+	if err != nil {
+		return errors.New("RemoveSourceInstallDirError: " + err.Error())
+	}
+
+	installDirStr := installedItem.InstallDirectory.String()
```

Comment:
installDir again, what is that? InstallDirectory is a field of the installedItem because it means the root dir of the item, the doc root. Therefore, the name here is itemDirStr ou itemRootDirStr dude.

Reply:
Done!